code/game/turfs/simulated/minerals.dm

```diff
@@ -20,37 +20,48 @@
 	color = COLOR_ROCK
 	var/environment_type = "asteroid"
 	var/turf/simulated/floor/plating/turf_type = /turf/simulated/floor/plating/asteroid/airless
-	var/mineralType = null
-	var/mineralAmt = 3
-	var/spread = 0 //will the seam spread?
-	var/spreadChance = 0 //the percentile chance of an ore spreading to the neighboring tiles
 	var/last_act = 0
-	var/scan_state = "" //Holder for the image we display when we're pinged by a mining scanner
+
 	var/defer_change = 0
 	var/mine_time = 4 SECONDS //Changes how fast the turf is mined by pickaxes, multiplied by toolspeed
 	/// Should this be set to the normal rock colour on init?
 	var/should_reset_color = TRUE
 
+	var/preset_ore_type
+	var/datum/ore/ore
+
 /turf/simulated/mineral/Initialize(mapload)
 	. = ..()
+
 	if(should_reset_color)
 		color = COLOR_ROCK
 
-	if(mineralType && mineralAmt && spread && spreadChance)
-		for(var/dir in GLOB.cardinal)
-			if(prob(spreadChance))
-				var/turf/T = get_step(src, dir)
-				if(istype(T, /turf/simulated/mineral/random))
-					Spread(T)
+	if(preset_ore_type)
+		set_ore(preset_ore_type)
+
 	AddComponent(/datum/component/debris, DEBRIS_ROCK, -20, 10, 1)
 
-/turf/simulated/mineral/proc/Spread(turf/T)
-	T.ChangeTurf(type)
+/turf/simulated/mineral/proc/set_ore(ore_type)
+	if(!ore_type)
+		return
+
+	ore = new ore_type()
+	if(ore.spread_chance)
+		for(var/dir in GLOB.cardinal)
+			if(prob(ore.spread_chance))
+				var/turf/simulated/mineral/T = get_step(src, dir)
+				if(istype(T))
+					T.set_ore(ore_type)
 
 /turf/simulated/mineral/shuttleRotate(rotation)
 	QUEUE_SMOOTH(src)
 
 /turf/simulated/mineral/attackby(obj/item/I, mob/user, params)
+	// TODO: Just sticking this here for now because attack chain refactor is coming later
+	// No point in threading this through everything right now
```

Comment:
TODO here

Reply:
Yup, as stated, I didn't pull this up the attack chain because I don't want to encourage anyone else to rely on it at the moment, and I can roll it into an attack chain refactor PR later on.